src/ecc.rs

```diff
@@ -174,78 +163,45 @@ impl Ecc {
     ) -> Result<Bytes> {
         let mut buf = BytesMut::with_capacity(ATCA_CMD_SIZE_MAX as usize);
         for retry in 0..retries {
-            buf.clear();
+            buf.clear();         
             command.bytes_into(&mut buf);
+            
+            self.transport.send_wake()?;
 
-            self.send_wake();
-            thread::sleep(WAKE_DELAY);
+            match self.transport.protocol {
+                TransportProtocol::I2c => {buf[0] = ATCA_I2C_COMMAND_FLAG}
+                TransportProtocol::Swi => {buf[0] = ATCA_SWI_COMMAND_FLAG}
+            }
 
-            if let Err(_err) = self.send_recv_buf(command.duration(), &mut buf) {
+            let delay = match self.transport.protocol {
+                TransportProtocol::I2c => {command.duration( false )}
+                TransportProtocol::Swi => {command.duration( true )}
+            };
+            
+            if let Err(_err) = self.transport.send_recv_buf(delay, &mut buf) {
                 if retry == retries {
                     break;
                 } else {
                     continue;
                 }
             }
 
-            let response = EccResponse::from_bytes(&buf[..])?;
+            let response = EccResponse::from_bytes(&buf[..]);
```

Comment:
Thinking further about this... This could be resolved by making a `CRCError` field in the `EccError` enum and return `Ok(Self::Error(EccError::CRCError))` similarly to how the other errors are handled in `from_bytes()` in command.rs